### PR TITLE
Active branch focused git.log() result.

### PR DESCRIPTION
**Pull request by codingtwinky**
fix: #420

~~Currently, what this fix is doing is that if "head" ref is missing from git.log, it will continuously increase git.log result list until it does.  Effectively always loading active git branch on load.
Now, this is not optimal and not my favorite.  I'm planning on writing more optimized solution as it can be bit slow and little barbaric solution as it is...~~

Added two config flags

* numberOfNodesPerLoad: number of nodes to load for each git.log call
* alwaysLoadActiveBranch: Always load with active checkout branch. ()

When `alwaysLoadActiveBranch=true`, terrible name but not sure what I can call it,  ungit will load checked out node on load by skipping previous nodes untill active checked out node is found for `git.log` call.  

I have also added a cheeky little button where you can load skipped nodes.

After more testing we might want to change the default value of `alwaysLoadActiveBranch` to true.

jrbixycto9

### CODE REVIEW

File: source/git-parser.js [hunk 1]
```diff
@@ -96,6 +96,8 @@ exports.parseGitLog = (data) => {
       const refs = row.substring(refStartIndex + 1, row.length - 1);
       currentCommmit.refs = refs.split(/ -> |, /g);
     }
+    currentCommmit.isHead = currentCommmit.refs.includes('HEAD');
+    commits.isHeadExist = commits.isHeadExist || currentCommmit.isHead;
```
Comment (codingtwinky) [inline source/git-parser.js]: Abusing javascripts aray flexibility!! yay!!!

**Comment by FredrikNoren** [comment]
Hey this is really cool! Really like it.

Two points:
1) The idea with the dotted circle was that it's the representation of the "next commit". It's kind of where the stuff in the staging area goes. How about something like this instead?:
collapse
2) Bug: Seems like creating a new branch isn't working while in the "collapsed" mode

**Comment by codingtwinky** [comment]
1. I may do that as a future improvement later when I stop drinking like a russian pirate 
2. Quick UI test by me had branch operations working even in collapsed mode.  Will add tests later

**Comment by codingtwinky** [comment]
merging for now.  

I really hope people who needed #420 experiment with this new feature and discuss how we can further improve this new feature.

**Pull request merged**